### ISSUE TITLE
Quasar false warning when using an object in v-for inside transition-group

### ISSUE THREAD

**Issue by asajadian** [quasarframework/quasar#7494]
**Describe the bug**
This is not exaclty a bug but a false warning showing up when an object is used in v-for inside a `transition-group`. The Quasar terminal output complains that index is used as the key, while the object property key is used!
`Do not use v-for index as key on <transition-group> children, this is the same as not using keys.`
here is the markup causing this

```
<transition-group
style="height: 100%"
class="flex items-center no-wrap"
enter-active-class="animated slideInDown "
leave-active-class="animated  slideOutDown"
>
  <MyComp
  v-for="(value, key) in object"
  :key="key"
  class="q-mx-xs"
  />
</transition-group>
```

**Expected behavior**
No warning to be shown

**Platform:**
Node: 10.16.3
NPM: 6.13.1
@quasar/app: 2.0.2
quasar cli: 1.1.0

**Comment by pdanpdan** [comment]
Hello. TransitionGroup is core Vue.

**Comment by hawkeye64** [comment]
@asajadian You are confused that `key` is not actually `index`. In the way you have used it, the second (optional) parameter is always `index`, regardless of the name you gave it -- in this case `key`. And `value` will be your object.
If you have an object, with `key`, then you have to destructure it. Let's say you also have property `name` in your object. Then,
```html
v-for="{ name, key } in object"
```

**Comment by asajadian** [comment]
@hawkeye64 Thanks for your input, excuse me I was away for a few days.
Maybe I did not give enough details what object is, the object I'm trying to loop through is like this:
```
const object = { 
  id321: { name: "john", age: "22"},
  id632: {name: "jack", age: "33"},
  id901: {name: "bob", age: 12}
}
```
Now if I do
` <p v-for="(value, key) in object" :key="key">{{key}}</p>`
The output is:
```
id321
id632
id901
```
So key is not actually an incrementing index, it is guaranteed to be unique. It seems Vue should make a distinction between objects and arrays and avoid giving this false warning.
But I found out that if I use `:key="key.toString()` the warning goes away and my terminal gets cleaned again!
